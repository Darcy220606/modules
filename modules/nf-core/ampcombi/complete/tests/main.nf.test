nextflow_process {

    name "Test Process AMPCOMBI_COMPLETE"
    script "../main.nf"
    process "AMPCOMBI_COMPLETE"

    tag "modules"
    tag "modules_nfcore"
    tag "ampcombi"
    tag "ampcombi/complete"

    config "./nextflow.config"
    setup {
        run("GUNZIP", alias:"GUNZIP_INPUT_GENOME_1") {
            script "../../../gunzip/main.nf"
                process {
                    """
                    input[0] =  Channel.fromList([
                        tuple([ id:'sample_1', single_end:false ], // meta map
                                file(params.test_data['bacteroides_fragilis']['illumina']['test1_contigs_fa_gz'], checkIfExists: true))
                    ])
                    """
            }
        }
        run("PYRODIGAL", alias:"PYRODIGAL_1") {
            script "../../../pyrodigal/main.nf"
            process {
                """
                input[0] = GUNZIP_INPUT_GENOME_1.out.gunzip
                input[1] = 'gbk'
                """
            }
        }
        run("GUNZIP", alias:"GUNZIP_GBK_1") {
            script "../../../gunzip/main.nf"
            process {
                """
                input[0] =  PYRODIGAL_1.out.annotations
                """
            }
        }
        run("GUNZIP", alias:"GUNZIP_FAA_1") {
            script "../../../gunzip/main.nf"
            process {
                """
                input[0] =  PYRODIGAL_1.out.faa
                """
            }
        }
        run("AMPIR", alias:"AMPIR_1") {
            script "../../../ampir/main.nf"
            process {
                """
                input[0] =  GUNZIP_FAA_1.out.gunzip
                input[1] = "precursor"
                input[2] = 10
                input[3] = "0.7"
                """
            }
        }
        run("MACREL_CONTIGS", alias:"MACREL_CONTIGS_1") {
            script "../../../macrel/contigs/main.nf"
            process {
                """
                input[0] = GUNZIP_INPUT_GENOME_1.out.gunzip
                """
            }
        }
        run("GUNZIP", alias:"GUNZIP_MACREL_1") {
            script "../../../gunzip/main.nf"
            process {
                """
                input[0] =  MACREL_CONTIGS_1.out.amp_prediction
                """
            }
        }
        run("AMPCOMBI_PARSETABLES", alias:"AMPCOMBI_PARSETABLES_1") {
            script "../../../ampcombi/parsetables/main.nf"
            process {
                """
                ch_amp_results = Channel.empty()
                ch_amp_results.mix(AMPIR_1.out.amps_tsv)
                ch_amp_results.mix(GUNZIP_MACREL_1.out.gunzip)
                ch_faa = GUNZIP_FAA_1.out.gunzip.map{ it[1] } //this does not require a meta
                ch_gbk = GUNZIP_GBK_1.out.gunzip.map{ it[1] } //this does not require a meta

                ch_input_parsetables = ch_amp_results
                    .groupTuple()
                    .join( ch_faa )
                    .join( ch_gbk )
                    .multiMap{
                        input: [ it[0], it[1] ]
                        faa: it[2]
                        gbk: it[3]
                    }

                input[0] = ch_input_parsetables.input
                input[1] = ch_input_parsetables.faa
                input[2] = ch_input_parsetables.gbk
                input[3] = []
                """
            }
        }
        run("GUNZIP", alias:"GUNZIP_INPUT_GENOME_2") {
            script "../../../gunzip/main.nf"
                process {
                    """
                    input[0] =  Channel.fromList([
                        tuple([ id:'sample_2', single_end:false ], // meta map
                                file(params.test_data['bacteroides_fragilis']['illumina']['test1_contigs_fa_gz'], checkIfExists: true))
                    ])
                    """
            }
        }
        run("PYRODIGAL", alias:"PYRODIGAL_2") {
            script "../../../pyrodigal/main.nf"
            process {
                """
                input[0] = GUNZIP_INPUT_GENOME_2.out.gunzip
                input[1] = 'gbk'
                """
            }
        }
        run("GUNZIP", alias:"GUNZIP_GBK_2") {
            script "../../../gunzip/main.nf"
            process {
                """
                input[0] =  PYRODIGAL_2.out.annotations
                """
            }
        }
        run("GUNZIP", alias:"GUNZIP_FAA_2") {
            script "../../../gunzip/main.nf"
            process {
                """
                input[0] =  PYRODIGAL_2.out.faa
                """
            }
        }
        run("AMPIR", alias:"AMPIR_2") {
            script "../../../ampir/main.nf"
            process {
                """
                input[0] =  GUNZIP_FAA_2.out.gunzip
                input[1] = "precursor"
                input[2] = 10
                input[3] = "0.7"
                """
            }
        }
        run("MACREL_CONTIGS", alias:"MACREL_CONTIGS_2") {
            script "../../../macrel/contigs/main.nf"
            process {
                """
                input[0] = GUNZIP_INPUT_GENOME_2.out.gunzip
                """
            }
        }
        run("GUNZIP", alias:"GUNZIP_MACREL_2") {
            script "../../../gunzip/main.nf"
            process {
                """
                input[0] =  MACREL_CONTIGS_2.out.amp_prediction
                """
            }
        }
        run("AMPCOMBI_PARSETABLES", alias:"AMPCOMBI_PARSETABLES_2") {
            script "../../../ampcombi/parsetables/main.nf"
            process {
                """
                ch_amp_results = Channel.empty()
                ch_amp_results.mix(AMPIR_2.out.amps_tsv)
                ch_amp_results.mix(GUNZIP_MACREL_2.out.gunzip)
                ch_faa = GUNZIP_FAA_2.out.gunzip.map{ it[1] } //this does not require a meta
                ch_gbk = GUNZIP_GBK_2.out.gunzip.map{ it[1] } //this does not require a meta

                ch_input_parsetables = ch_amp_results
                    .groupTuple()
                    .join( ch_faa )
                    .join( ch_gbk )
                    .multiMap{
                        input: [ it[0], it[1] ]
                        faa: it[2]
                        gbk: it[3]
                    }

                input[0] = ch_input_parsetables.input
                input[1] = ch_input_parsetables.faa
                input[2] = ch_input_parsetables.gbk
                input[3] = []
                """
            }
        }
    }



    test("bacteroides_fragilis - genome") {
        when {
            process {
                """
                ch_amp_complete = Channel.empty()
                ch_amp_complete.mix(AMPCOMBI_PARSETABLES_1.out.tsv.map{ it[1] })
                ch_amp_complete.mix(AMPCOMBI_PARSETABLES_2.out.tsv.map{ it[1] })

                input[0] = ch_amp_complete
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("bacteroides_fragilis - genome - stub") {
        options "-stub"
        when {
            process {
                """
                ch_amp_complete = Channel.empty()
                ch_amp_complete.mix(AMPCOMBI_PARSETABLES_1.out.tsv.map{ it[1] })
                ch_amp_complete.mix(AMPCOMBI_PARSETABLES_2.out.tsv.map{ it[1] })

                input[0] = ch_amp_complete
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
