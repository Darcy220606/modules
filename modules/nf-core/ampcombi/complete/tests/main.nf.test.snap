{
    "bacteroides_fragilis - genome - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "log": [
                    
                ],
                "tsv": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-26T09:56:37.164199088"
    },
    "bacteroides_fragilis - genome": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "log": [
                    
                ],
                "tsv": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-26T09:56:04.779714294"
    }
}